### PR TITLE
Correct/remove timezone

### PR DESCRIPTION
**Pull request by Shylock-Hg**

### CODE REVIEW

**Review by yixinglu — APPROVED** [review]

**Review by CPWstatic — DISMISSED** [review]

File: src/optimizer/OptimizerUtils.cpp [hunk 1]
```diff
@@ -274,7 +274,15 @@ Value OptimizerUtils::boundValueWithMax(const meta::cpp2::ColumnDef& col, const
             d.day = 31;
             return Value(d);
         }
-        case Value::Type::DATETIME : {
+        case Value::Type::TIME: {
+            Time dt;
+            dt.hour = 24;
```
Comment (laura-ding) [inline src/optimizer/OptimizerUtils.cpp:279]: 23

File: src/optimizer/OptimizerUtils.cpp [hunk 2]
```diff
@@ -274,7 +274,15 @@ Value OptimizerUtils::boundValueWithMax(const meta::cpp2::ColumnDef& col, const
...
+            dt.minute = 60;
```
Comment (laura-ding) [inline src/optimizer/OptimizerUtils.cpp:280]: 59

File: src/optimizer/OptimizerUtils.cpp [hunk 3]
```diff
@@ -274,7 +274,15 @@ Value OptimizerUtils::boundValueWithMax(const meta::cpp2::ColumnDef& col, const
...
+            dt.sec = 60;
```
Comment (laura-ding) [inline src/optimizer/OptimizerUtils.cpp:281]: 59

File: src/optimizer/OptimizerUtils.cpp (hunk 1 above)
Comment (bright-starry-sky) [reply src/optimizer/OptimizerUtils.cpp:279]: > 23, the DateTime ditto

Thanks for your remind. I think the 24, 60 , 60 is correct. if we want range scan, the max value should be max + 1. for example :
rang(1, 3) , the result is 1, 2. so if we want result is 1,2,3 , the range pair must be {1, 4} .

File: src/optimizer/OptimizerUtils.cpp [hunk 4]
```diff
@@ -311,12 +332,28 @@ Value OptimizerUtils::boundValueWithMin(const meta::cpp2::ColumnDef& col, const
...
+        case Value::Type::SET:
+        case Value::Type::MAP:
+        case Value::Type::DATASET:
+        case Value::Type::PATH: {
+            DLOG(FATAL) << "Not supported value type " << type
+                        << "for index.";
             return Value(NullType::BAD_TYPE);
+        }
     }
+    DLOG(FATAL) << "Unknown value type " << static_cast<int>(type);
```
Comment (bright-starry-sky) [inline src/optimizer/OptimizerUtils.cpp:355]: It's better not to add FATAL at here.

Comment (Shylock-Hg) [reply src/optimizer/OptimizerUtils.cpp:355]: It's exception that the type is never defined.

File: src/optimizer/OptimizerUtils.cpp [hunk 5]
```diff
@@ -311,12 +332,28 @@ Value OptimizerUtils::boundValueWithMin(const meta::cpp2::ColumnDef& col, const
...
-        default :
+        case Value::Type::__EMPTY__:
+        case Value::Type::NULLVALUE:
+        case Value::Type::VERTEX:
+        case Value::Type::EDGE:
+        case Value::Type::LIST:
+        case Value::Type::SET:
+        case Value::Type::MAP:
+        case Value::Type::DATASET:
+        case Value::Type::PATH: {
```
Comment (bright-starry-sky) [inline src/optimizer/OptimizerUtils.cpp:349]: What purpose? Identify missing types by compilation error?

Comment (Shylock-Hg) [reply src/optimizer/OptimizerUtils.cpp:349]: Yes.

**Review by bright-starry-sky — COMMENTED** [review]
If you want add the new type of Time for `OptimizerUtils.cpp`, please help change the method `boundValueWithLT` and `boundValueWithGT` together. thanks.

**Comment by bright-starry-sky** [comment]
> If you want add the new type of Time for `OptimizerUtils.cpp`, please help change the method `boundValueWithLT` and `boundValueWithGT` together. thanks.

Or I'll take care of it later in other PR. 
Good job. Thanks.

**Comment by Shylock-Hg** [comment]
> > If you want add the new type of Time for `OptimizerUtils.cpp`, please help change the method `boundValueWithLT` and `boundValueWithGT` together. thanks.
> 
> Or I'll take care of it later in other PR.
> Good job. Thanks.

Had done, Good catch.

File: src/optimizer/OptimizerUtils.cpp [hunk 6]
```diff
@@ -202,6 +239,33 @@ Value OptimizerUtils::boundValueWithLT(const meta::cpp2::ColumnDef& col, const V
             }
             return Value(d);
         }
+        case Value::Type::TIME : {
+            if (DateTime() == v.getDateTime()) {
+                return v.getDateTime();
+            }
+            auto dt = v.getDateTime();
```
Comment (dutor) [inline src/optimizer/OptimizerUtils.cpp:246]: `getTime`?

File: src/optimizer/OptimizerUtils.cpp [hunk 7]
```diff
@@ -94,6 +94,31 @@ Value OptimizerUtils::boundValueWithGT(const meta::cpp2::ColumnDef& col, const V
             }
             return Value(d);
         }
+        case Value::Type::TIME : {
+            auto dt = v.getDateTime();
```
Comment (dutor) [inline src/optimizer/OptimizerUtils.cpp:98]: ditto.

File: src/optimizer/OptimizerUtils.cpp (hunk 6 above)
Comment (Shylock-Hg) [reply src/optimizer/OptimizerUtils.cpp]: Good catch. Fixed.

**Review by dutor — APPROVED** [review]

**Review by laura-ding — APPROVED** [review]

**Pull request merged**